src/encoded/static/components/genome_browser.js

```diff
@@ -0,0 +1,432 @@
+                this.browserFiles.push({
+                    name: trackLabels.shortLabel,
+                    desc: trackLabels.longLabel,
+                    bwgURI: `${domain}${file.href}`,
+                    style: [
+                        {
+                            type: 'default',
+                            style: {
+                                glyph: 'HISTOGRAM',
+                                HEIGHT: 30,
+                                BGCOLOR: 'rgb(166,71,71)',
+                            },
+                        },
+                    ],
+                });
+            } else if (file.file_format === 'bigBed') {
+                this.browserFiles.push({
+                    name: trackLabels.shortLabel,
+                    desc: trackLabels.longLabel,
+                    bwgURI: `${domain}${file.href}`,
+                    style: [
+                        {
+                            style: {
+                                HEIGHT: 10,
+                            },
+                        },
+                    ],
+                });
+            }
+        });
+        if (this.browserFiles.length) {
+            browserCfg.sources = browserCfg.sources.concat(this.browserFiles);
+        }
+
+        require.ensure(['dalliance'], (require) => {
+            const Dalliance = require('dalliance').browser;
+
+            this.browser = new Dalliance({
+                maxHeight: 1000,
+                // noPersist: true,
+                // noPersistView: true,
+                noTrackAdder: true,
+                maxWorkers: 4,
+                noHelp: true,
+                chr: browserCfg.chr,
+                viewStart: browserCfg.viewStart,
+                viewEnd: browserCfg.viewEnd,
+                cookieKey: browserCfg.cookieKey,
+                coordSystem: browserCfg.coordSystem,
+                sources: browserCfg.sources,
+                noTitle: true,
+                disablePoweredBy: true,
+            });
+        });
+    },
+
+    componentDidUpdate: function () {
+        // Remove old tiers
+        if (this.browser && this.browserFiles && this.browserFiles.length) {
+            this.browserFiles.forEach((fileSource) => {
+                this.browser.removeTier({
+                    name: fileSource.name,
+                    desc: fileSource.desc,
+                    bwgURI: fileSource.bwgURI,
+                });
+            });
+        }
+
+        const files = !this.context.localInstance ? this.props.files.slice(0, maxFilesBrowsed - 1) : dummyFiles;
+        if (this.browser && files && files.length) {
+            let domain = `${location.protocol}//${location.hostname}`;
+            if (domain.includes('localhost')) {
+                domain = domainName;
+            }
+            files.forEach((file) => {
+                const trackLabels = this.makeTrackLabel(file);
+                if (file.file_format === 'bigWig') {
```

Comment:
That just has to be factored out. Like with home-page charts, you have to do the same thing on initial draw as well as update. I factored that out on the home page, but haven’t touched this one since being in desperation mode.